app/grandchallenge/notifications/views.py

```diff
@@ -1,27 +1,113 @@
+        if "delete" in request.POST:
+            action = "delete"
+        elif "mark_read" in request.POST:
+            action = "mark_read"
+        elif "mark_unread" in request.POST:
+            action = "mark_unread"
+
+        selected_notifications = request.POST.getlist("checkbox")
+        notifications = get_objects_for_user(
+            request.user,
+            ["delete_notification", "change_notification"],
+            Notification,
+        ).filter(id__in=selected_notifications)
+        notifications_count = notifications.count()
+        if not notifications:
+            return HttpResponseNotFound()
+        else:
+            if action == "delete":
+                notifications.delete()
+                messages.add_message(
+                    request,
+                    messages.SUCCESS,
+                    f"{notifications_count} notification{pluralize(notifications_count)} successfully deleted.",
+                )
+            elif action == "mark_read":
+                notifications.update(read=True)
+                messages.add_message(
+                    request,
+                    messages.SUCCESS,
+                    f"{notifications_count} notificiation{pluralize(notifications_count)} successfully marked as read.",
+                )
+            elif action == "mark_unread":
+                notifications.update(read=False)
+                messages.add_message(
+                    request,
+                    messages.SUCCESS,
+                    f"{notifications_count} notification{pluralize(notifications_count)} successfully marked as unread.",
+                )
+            return HttpResponseRedirect(reverse("notifications:list"))
 
-        context.update(
-            {
-                "notifications_last_read_at": profile.notifications_last_read_at,
-                "object_list": profile.notifications,
-            }
-        )
 
-        UserProfile.objects.filter(pk=profile.pk).update(
-            notifications_last_read_at=now()
+class FollowList(LoginRequiredMixin, PermissionListMixin, ListView):
+    model = Follow
+    permission_required = "view_follow"
+
+    def get_queryset(self, *args, **kwargs):
+        return prefetch_generic_foreign_key_objects(
+            super().get_queryset().select_related("user")
         )
 
-        return context
+
+class FollowDelete(
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    SuccessMessageMixin,
+    DeleteView,
+):
+    model = Follow
+    success_message = "Subscription successfully deleted"
+    permission_required = "delete_follow"
+    raise_exception = True
+
+    def get_permission_object(self):
+        return self.get_object()
+
+    def get_success_url(self):
+        return reverse("notifications:follow-list")
+
+
+class FollowCreate(
+    LoginRequiredMixin, SuccessMessageMixin, CreateView,
+):
+    model = Follow
+    fields = ["user", "content_type", "object_id", "actor_only"]
```

Comment:
I think that this view might be too permissive. I think that this means that any user (including anonymous users) can create a follow for any other user, as there is no validation that the `user` provided is the current user from the request. We should also check that the user has permission to follow that object. Some things might be private and information could be leaked if this is not checked carefully, it might need separate methods as this is different for Forums, Topics, and any other objects we add in the future (eg, evaluations).